service/storage_proxy.cc

```diff
@@ -2763,18 +2763,21 @@ class abstract_read_resolver {
         _timeout.arm(timeout);
     }
     virtual ~abstract_read_resolver() {};
-    virtual void on_error(gms::inet_address ep, bool disconnect) = 0;
+    virtual void on_error(gms::inet_address ep, bool disconnect, std::exception_ptr overloaded) = 0;
```

Comment:
Instead of passing a pointer to an exception here, why not change `bool disconnect` to `error_type type` instead where error_type can be `enum {disconnect, overloaded}`

Reply:
I used the exception pointer since I already had it from catching the exception, but an enum also makes sense. I'll soon push an unrelated PR which adds an error enum to storage proxy anyway, so I'll wait for that one to be reviewed first. If it goes in, I can just rebase and use the already existing enum.